### ISSUE TITLE
Startup problem with 10/23/204 release:  tries to run INTERPRET.REM.CM again

### ISSUE THREAD

**Issue by rmkaplan** [Interlisp/medley#1863]
**Describe the bug**
At startup, the full.sysout runs the usual greeting (system and user), but then fails because it can't find MEDLEY-DIR.LCOM.  It appears to be running INTERPRET.REM.CM, which is trying to run the LDEINIT file.

**To Reproduce**
Start up the sysout in the usual way

**Expected behavior**
Greeting completes successfully

**Context (please complete the following information):**
 - OS: Mac
 - OS Sonoma 14.6.1
 - M2 chip
 - Maiko version: [e.g. commit ID from `git log | head`]
 - IL:MAKESYSDATE: [ 23-Oct-2024 02:24:13 ]

**Comment by nbriggs** [comment]
I can't reproduce this.  I made a `full.sysout` on the rmk36--Tedit-fifth-round-2 branch (23 October 2024), then removed my ldeinit so that it wouldn't find it if it tried, and then started medley using

```
cd medley && ./run-medley   -g 1200x700 -sc 1200x700 -noscroll loadups/full.sysout
```
and it started and loaded the MEDLEY-DIR.LCOM and showed no errors.

**Comment by rmkaplan** [comment]
I saw this in the release sysout, not one in my branches or that I constructed.

**Comment by masinter** [comment]
the initialization is complex because there are a lot of possibilities and the details matter.
The transcript from the shell when running the 'medley' script is useful to diagnose:

```
"/home/larry/il/medley/../maiko/linux.x86_64/lde" "/home/larry/il/medley/loadups/full.sysout" -id "default" -title "Medley Interlisp " -g 1880x1016 -sc 1880x1016 -noscroll
MEDLEYDIR: "/home/larry/il/medley"
LOGINDIR: "/home/larry/il"
GREET FILE: "/home/larry/il/medley/greetfiles/MEDLEYDIR-INIT.LCOM"
VMEM FILE: "/home/larry/il/vmem/lisp.virtualmem"
requested width: 1888, height: 1016
initialised
Window created
Creating renderer...
Creating texture...
SDL initialised
Unsupported keyboard type: 14
Configuring keyboard for type-3
```

in particular, what is GREETFILE (= LDEINIT) and the initial content of it ... INTERPRET.REM.CM is called when the contents of $LDEINIT starts with a " double quote, not a "(" left paren

**Comment by rmkaplan** [comment]
Did you see the same problem, starting up the release system from last Monday/Tuesday ?

**Comment by masinter** [comment]
i'm saying I didn't see the problem and can't reproduce it, starting from the latest release, and need more info to give any advice about what's going on.

**Comment by rmkaplan** [comment]
I tried it again, in the most recent release.  I downloaded the loadups.tgz and runtime.tgz, merged them.  Then ran

kaplan@Ron-Macbook-Air git-medley %  rmedley
"/Users/kaplan/Local/medley3.5/release-medley/../maiko/darwin.aarch64/lde" "/Users/kaplan/Local/medley3.5/release-medley/loadups/full.sysout" -id "1" -title "Release Medley:: 1" -g 1920x1036 -sc 1888x1004
MEDLEYDIR: "/Users/kaplan/Local/medley3.5/release-medley"
LOGINDIR: "/Users/kaplan/lisp"
GREET FILE: "/Users/kaplan/Local/medley3.5/release-medley/greetfiles/MEDLEYDIR-INIT.LCOM"
VMEM FILE: "/Users/kaplan/lisp/vmem/lisp_1.virtualmem"

These screen shots show the beginning of the greeting and the break at the end. Below I'm also providing my rmedley script
![Screenshot 2024-12-27 at 11 48 06 AM](https://github.com/user-attachments/assets/d9547066-e17b-47d3-80ec-9817f39c884b)

![Screenshot 2024-12-27 at 11 48 29 AM](https://github.com/user-attachments/assets/8776431a-b534-48ee-b0dc-ad51da549d6a)

export medleyc="/scripts/medley/medley.command $screen -i +"
alias rmedley=' /bin/sh $RMEDLEYDIR/$medleyc  -t Release\ Medley%i --full '

This is the way I'm calling Frank's medleyc when I start up the Medley clone my working directory loadups

**Comment by rmkaplan** [comment]
A propos of the meeting today, I downloaded the mac universal and ran ./medley directly in the medley/ subdirectory.  The presumably runs the maiko that is part of the download.

I'll note as a first observation that I did not run the xattr for maiko/darwin-universal/lde* that is specified in the How to run on Mac web page, and the system came up as expected.  So it is not clear that that particular xattr is necessary.  Should that instruction be removed?

I ran into some other problems with my personal Init, because this configuration of files is not located properly relative to the working directory that my init also looks for.  So I have to make some other adjustments before I can verify whether or not the MEDLEYDIR-INIT is being run twice.

**Comment by rmkaplan** [comment]
Some more information, again running ./medley at the medley/ subdirectory of the univeral-mac download, with nothing else changed.  No xattr's done.

Here is the screen image at the beginning of the startup.  Note that it is in the middle of running the MEDLEYDIR-INIT.LCOM from the downloaded directory

![Screenshot 2024-12-30 at 3 01 42 PM](https://github.com/user-attachments/assets/a99a57a4-6325-4a10-8d4e-77de93c57524)

I put in HELP's at the beginning and end of my personal INIT.  In the HELP, I confirmed the existence of the MEDLEYDIR-INIT.LCOM file:
![Screenshot 2024-12-30 at 3 05 53 PM](https://github.com/user-attachments/assets/49343c94-fd2b-4c09-9de5-2cd54bad773a)

It ran thru to the HELP at the end (stumbling a little bit on some personal things that it can't find from the downloaded location).  And then it finished my personal init...and tried again for the MEDLEYDIR-INIT.LCOM.  I broke it while it was doing the exchange pups as part of a spell-correction for that unfound-file:

![Screenshot 2024-12-30 at 3 08 05 PM](https://github.com/user-attachments/assets/8af234ba-67fc-4cdc-ad39-40520e8b49f0)

So: 

Should it have been trying again, after it has already done the site greeting and my personal init?

If so, why cannot it not find the file the second time when it could find and execute it the first time?

**Comment by rmkaplan** [comment]
(Note BTW that MEDLEYDIR-INIT.LCOM is the name of the file in the OPENSTREAM under INTERPRET.REM.CM

**Comment by masinter** [comment]
The GREET code as currently defined will try to read the file named by LDEINIT (a unix file name but treated as {DSK}). Maybe changing LDEINIT to explicitly have a {DSK}?
The quarantined files -- are they INFILEP? Even when they give an error when you try to OPENFILE?

**Comment by rmkaplan** [comment]
MEDLEYDIR-INIT.LCOM is must be INFILEP becaise it is read to start the greeting (presumably because it is the value of LDEINIT (without {DSK}).  But it is apparently not INFILEP when it breaks on the second attempt, after my personal init has completed.

**Comment by rmkaplan** [comment]
Have you tried this on your mac?  Run ./medley in the medley/ download of the mac universal.

**Comment by nbriggs** [comment]
When I try it on macOS Ventura 13.7.1, unpacking into `/tmp/241223/` this is what I see:
![Screenshot 2024-12-31 at 10 28 24 AM](https://github.com/user-attachments/assets/406e8cea-1939-4b2a-a298-a4e25b734c44)

**Comment by rmkaplan** [comment]
Did you have a personal greeting file?

**Comment by nbriggs** [comment]
`(GREETFILENAME 'briggs)` doesn't think so.

**Comment by nbriggs** [comment]
I'm pretty sure it's running it twice - but my macOS version is different from yours and so the file protection code will be different (and I don't have Terminal set to have full-disk-access, though I have some extra files & folders open for it).

You could try doing "sudo fs_usage ldex >/tmp/fs" in one terminal window before starting medley, and see if you see any protection issues show up there.

**Comment by MattHeffron** [comment]
It occurred to me that a `~/.medley_config` file that sets the greet file (i.e., -r FILE) could be complicating diagnosing this. 
> Maybe changing LDEINIT to explicitly have a {DSK}?

This could be tricky; because, if provided (command line or config), as it is getting parsed, it is checked by `medley.command` to see if it is a readable file. So, the `{DSK}` would need to be added _after_ that check.

**Comment by rmkaplan** [comment]
This seems to work as expected in the 250127 release.

**Comment by pamoroso** [comment]
I'm closing the issue as the system now [seems to work as expected](https://github.com/Interlisp/medley/issues/1863#issuecomment-2623447933).
